soco/services.py

```diff
@@ -283,6 +314,71 @@ def unwrap_arguments(xml_response):
             "{http://schemas.xmlsoap.org/soap/envelope/}Body")[0]
         return dict((i.tag, i.text or "") for i in action_response)
 
+    def compose_args(self, action_name, in_arglist, in_argdict):
+        """Compose the argument list from list of tuples or dictionary form.
+
+        Args:
+            action_name (str): The name of the action to be performed.
+            in_arglist (list): Arguments as a list of ``(name, value)``
+                tuples specifying the name of each argument and its value, eg
+                ``[('InstanceID', 0), ('Speed', 1)]``. The value
+                can be a string or something with a string representation.
+            in_argdict (dict): Arguments as a dict, eg
+                ``{'InstanceID': 0, 'Speed': 1}.
+
+        This merges args and kwargs into a single argument list, with
+        ``DEFAULT_ARGS`` as default values.
+        The ``argdict`` takes precedence over the ``arglist``.
+
+        Returns:
+            list: a list of ``(name, value)`` tuples.
+
+        Raises:
+            `AttributeError`: If this service does not support the action.
+            `ValueError`: If the argument lists do not match the action
+                signature.
+        """
+        in_argdict2 = dict(in_arglist or ())
```

Comment:
not a big fan of `foo_bar2`, how about `in_argdict_from_arglist`?

Reply:
I'll go with `in_arglist_as_dict`.

Reply:
Absolutely, "short, descriptive and to the point" to quote Linuxs Thorvalds on naming git branches, variables and functions